fix: Fix rebalancing related tests

GigaCronos opened the pull request:
#### This commit fix the following tests by adjusting parameters:
- `test_chunk_rebalancing`
- `test_custom_goals_generate_no_accidental_rebalancing`

The other rebalancing tests are not failing.

Modified `chunks.cc` to allow small parameters for `HashCPS`, specifically for testing purposes.
Modified the Jenkinsfile to create a new test suite for testing with loop devices (focusing on rebalance for now).

#### **The tests were failing due to mechanisms that cause the `ChunkWorker` rebalancing logic to use outdated disk usage data:**
- Polling Delay (50ms): The Master's disk usage info is not real-time, causing decisions based on old states.
- Processing Snapshot: The `ChunkWorker::mainLoop` uses a fixed disk usage snapshot for up to 1 second. During this time, its own actions change the real state, making subsequent decisions increasingly inaccurate.

**Test Impact:** This data staleness creates false imbalance readings. One test fails due to imprecise rebalancing, and the other fails from triggering unnecessary operations based on outdated data.

Co-authored-by: Urmas Rist <urmas@leil.io>

GigaCronos committed:
fix: Fix rebalancing related tests

This commit fix the following tests by adjusting parameters:
- test_chunk_rebalancing
- test_custom_goals_generate_no_accidental_rebalancing

The other rebalancing tests don't have errors.

Modifies chunk.cc allowing small parameters for CPS,
only for testing purposes.

The main reason why these tests were failing is because
rebalancing decisions are made using outdated disk_usage
information, triggering unnecessary replication and
non optimal deletions.

Signed-off-by: GigaCronos <jorge.cabrera@leil.io>

GigaCronos committed:
test: Add Jenkins support for rebalancing tests.

Create new test suite, for testing with loop devices(only
rebalance for now).

Co-authored-by: Urmas Rist <urmas@leil.io>

Signed-off-by: GigaCronos <jorge.cabrera@leil.io>

antuan96314 commented on Jenkinsfile:
```diff
@@ -180,15 +180,15 @@ pipeline {
         stage('Build and test') {
             parallel {
-                stage('Build and run unittests (Ubuntu 24.04)') {
+                stage('Run special tests (Ubuntu 24.04)') {
```
Why is this needed?

uristdwarf replied:
This is because the tests don't work with the existing sfstests tool, due to requiring access to creating and using loop devices, which are kernel level. So it cannot be run in parallel and hard to get it working on Docker (I think it's possible, but I haven't managed it).

antuan96314 commented on Jenkinsfile:
```diff
@@ -180,15 +180,15 @@ pipeline {
                             }
                         }
-                        stage("Build and test") {
+                        stage("Build and unit test") {
```
Why is this needed?

uristdwarf replied:
See above

GigaCronos committed:
refactor: Apply AI suggestions

Sorted alphabetically the list of configuration options for better readability and maintainability.

Signed-off-by: GigaCronos <jorge.cabrera@leil.io>

GigaCronos commented on tests/test_suites/RebalancingTests/test_chunk_rebalancing.sh:
```diff
@@ -8,9 +8,11 @@ CHUNKSERVERS=5 \
 			`|OPERATIONS_DELAY_INIT = 0`
 			`|OPERATIONS_DELAY_DISCONNECT = 0`
+			`|POLL_TIMEOUT_MS = 10`
 			`|ACCEPTABLE_DIFFERENCE = 0.0015" \
```
Done

rolysr reviewed (approved):
LGTM.
